spec/index.bs

```diff
@@ -1835,27 +1832,87 @@ dictionary GPURenderBundleEncoderDescriptor : GPUObjectDescriptorBase {
 Queues {#queues}
 ================
 
+<script type=idl>
+dictionary GPUTextureDataLayout {
+    GPUBufferSize offset = 0;
+    required unsigned long rowPitch;
+    required unsigned long imageHeight;
+};
+</script>
+
 <script type=idl>
 interface GPUQueue {
     void submit(sequence<GPUCommandBuffer> commandBuffers);
 
     GPUFence createFence(optional GPUFenceDescriptor descriptor = {});
     void signal(GPUFence fence, unsigned long long signalValue);
 
+    void writeBuffer(
+        ArrayBuffer sourceData,
+        GPUBufferSize sourceOffset,
+        GPUBuffer destination,
+        GPUBufferSize destinationOffset,
+        GPUBufferSize size);
+
+    void writeTexture(
+        ArrayBuffer sourceData,
+        GPUTextureDataLayout sourceLayout,
+        GPUTextureCopyView destination,
+        GPUExtent3D size);
+
     void copyImageBitmapToTexture(
         GPUImageBitmapCopyView source,
         GPUTextureCopyView destination,
-        GPUExtent3D copySize);
+        GPUExtent3D size);
 };
 GPUQueue includes GPUObjectBase;
 </script>
 
- - {{GPUQueue/copyImageBitmapToTexture()}}:
-   - For now, `copySize.z` must be `1`.
+<dl dfn-type="method" dfn-for="GPUQueue">
+    : <dfn>writeBuffer(sourceData, sourceOffset, destination, destinationOffset, size)</dfn>
+    ::
+        Takes the `sourceData` contents starting from the offset `sourceOffset` and
+        schedules a write operation of these contents to the `destination` buffer in the queue.
+        Any subsequent modifications to sourceData do not affect what is written
+        at the time that the scheduled operation runs.
+        The operation does nothing and produces an error if any of the following is true:
+
+         - `destination` buffer wasn't created with {{GPUBufferUsage/COPY_DST}} flag in {{GPUBufferDescriptor/usage}}.
+         - `destination` buffer must not be destroyed
```

Comment:
```suggestion
         - `destination` buffer is destroyed
```

Reply:
Actually, this is implied by ["destination" buffer isn't in the "unmapped" state.], so you could also remove this.